code/datums/brain_damage/split_personality.dm

```diff
@@ -56,34 +52,28 @@
 		switch_personalities(TRUE)
 	QDEL_NULL(stranger_backseat)
 	QDEL_NULL(owner_backseat)
+	UnregisterSignal(owner, COMSIG_LIVING_DEATH)
 	..()
 
-/datum/brain_trauma/severe/split_personality/Destroy()
-	if(stranger_backseat)
-		QDEL_NULL(stranger_backseat)
-	if(owner_backseat)
-		QDEL_NULL(owner_backseat)
-	return ..()
+/datum/brain_trauma/severe/split_personality/proc/revert_to_normal()
+	qdel(src)
 
-/datum/brain_trauma/severe/split_personality/proc/switch_personalities(reset_to_owner = FALSE)
-	if(QDELETED(owner) || QDELETED(stranger_backseat) || QDELETED(owner_backseat))
+/datum/brain_trauma/severe/split_personality/proc/switch_personalities(forced = FALSE)
+	if(QDELETED(owner) || (owner.stat == DEAD && !forced) || QDELETED(stranger_backseat) || QDELETED(owner_backseat))
 		return
 
 	var/mob/living/split_personality/current_backseat
-	var/mob/living/split_personality/new_backseat
-	if(current_controller == STRANGER || reset_to_owner)
-		current_backseat = owner_backseat
-		new_backseat = stranger_backseat
-	else
+	var/mob/living/split_personality/free_backseat
+	if(current_controller == OWNER)
 		current_backseat = stranger_backseat
-		new_backseat = owner_backseat
-
-	if(!current_backseat.client) //Make sure we never switch to a logged off mob.
-		return
+		free_backseat = owner_backseat
+	else
+		current_backseat = owner_backseat
+		free_backseat = stranger_backseat
 
 	log_game("[key_name(current_backseat)] assumed control of [key_name(owner)] due to [src]. (Original owner: [current_controller == OWNER ? owner.key : current_backseat.key])")
-	to_chat(owner, span_userdanger("You feel your control being taken away... your other personality is in charge now!"))
-	to_chat(current_backseat, span_userdanger("You manage to take control of your body!"))
+	to_chat(owner, "<span class='userdanger'>You feel your control being taken away... your other personality is in charge now!</span>")
+	to_chat(current_backseat, "<span class='userdanger'>You manage to take control of your body!</span>")
```

Comment:
you must put skyrat edit

Reply:
No, this is incorrect. We do not use <span class=''>
only use span_notice() etc.